DatadogRUM/Tests/RUMMonitor/Scopes/RUMSessionScopeTests.swift

```diff
@@ -628,9 +628,55 @@ class RUMSessionScopeTests: XCTestCase {
             with `RUMMonitor.shared().startView()` and `RUMMonitor.shared().stopView()`.
             """
         )
+    }
+
+    func testGivenSessionWithNoActiveScope_whenReceivingSessionAliveCommand_itDoesNotLogWarning() throws {
+        func recordWarningOnReceiving(command: RUMCommand) -> String? {
+            // Given
+            let scope: RUMSessionScope = .mockWith(
+                parent: parent,
+                startTime: Date()
+            )
+            XCTAssertEqual(scope.viewScopes.count, 0)
+
+            let dd = DD.mockWith(logger: CoreLoggerMock())
+            defer { dd.reset() }
+
+            // When
+            _ = scope.process(command: command, context: context, writer: writer)
+
+            // Then
+            XCTAssertEqual(scope.viewScopes.count, 0)
+            return dd.logger.warnLog?.message
+        }
 
         let keepAliveCommand = RUMKeepSessionAliveCommand(time: Date(), attributes: [:])
         let keepAliveLog = recordWarningOnReceiving(command: keepAliveCommand)
         XCTAssertNil(keepAliveLog, "It shouldn't log warning when receiving `RUMKeepSessionAliveCommand`")
     }
+
+    func testGivenSessionWithNoActiveScope_whenReceivingPerformanceMetricCommand_itDoesNotLogWarning() throws {
+        func recordWarningOnReceiving(command: RUMCommand) -> String? {
+            // Given
+            let scope: RUMSessionScope = .mockWith(
+                parent: parent,
+                startTime: Date()
+            )
+            XCTAssertEqual(scope.viewScopes.count, 0)
+
+            let dd = DD.mockWith(logger: CoreLoggerMock())
+            defer { dd.reset() }
+
+            // When
+            _ = scope.process(command: command, context: context, writer: writer)
+
+            // Then
+            XCTAssertEqual(scope.viewScopes.count, 0)
+            return dd.logger.warnLog?.message
+        }
```

Comment:
nit/ The `recordWarningOnReceiving()` seems to be copy & pasted from previous test. We could reuse it this way:
```swift
func testGivenSessionWithNoActiveScope_whenReceivingCommands_itDoesNotLogWarning() throws {
    func recordWarningOnReceiving(command: RUMCommand) -> String? {
       // ...
    }

    // When
    let keepAliveCommand = RUMUpdatePerformanceMetric(...)
    let keepAliveLog = ...

    // Then
    XCTAssertNil(keepAliveLog, ...)

    // When
    let perfMetricCommand = RUMKeepSessionAliveCommand(...)
    let perfMetricLog = ...

    // Then
    XCTAssertNil(keepAliveLog, ...)
}
```

Reply:
It is copied yes, and I guess it's a style thing. I prefer a smaller number of assertions per test, but I'm good with combining them.  Maybe to make it cleaner or more "single responsibility" we could do:
```swift
let quietCommands = [ RumUpdatePerformanceMetric(...), RUMKeepSessionAliveCommand(...) ]

for command in quietCommands {
  let log = ...
  XCTAssertNil(log, ...)
}
```

What do you think?

Reply:
Sounds good 👌👍. In my opinion, the key is maintaining test clarity and keeping the number of tests manageable to avoid bloating this file. The loop should handle it 👍. `RUM*ScopeTests` often grow large since `RUM*Scopes` mix dozens of responsibilities 😞.